### PR TITLE
More datatypes work

### PR DESCRIPTION
**Pull request by jhdulaney**
## Description
Adds the auto_add_datatypes field for grandfathering in existing projects, adds a new API endpoint for displaying datatypes, and does some minor tweaks.

## Related Issue
#981 

## Testing
Set the auto_add_datatypes boolean via admin interface
Uploaded a datafile without datatype specified.
Verified that the datafile had the full set of project datatypes associated with it.

### CODE REVIEW

File: private_sharing/api_views.py [hunk 1]
```diff
@@ -254,6 +264,9 @@ def save_datatypes(self, data_file):
 
         datatypes can be looked up either via name or ID
         """
+        if self.project.auto_add_datatypes:
```
Comment (madprime) [inline private_sharing/api_views.py]: I think we should do this only if datatypes weren't specified.

File: private_sharing/serializers.py [hunk 2]
```diff
@@ -116,3 +116,23 @@ def to_representation(self, obj):
             rep.pop("username")
 
         return rep
+
+
+class DataTypeSerializer(serializers.ModelSerializer):
+    """
+    Serialize data for a project member.
```
Comment (madprime) [inline private_sharing/serializers.py]: incorrect due to copy/paste?

File: private_sharing/serializers.py [hunk 3]
```diff
@@ -116,3 +116,23 @@ def to_representation(self, obj):
...
+    class Meta:  # noqa: D101
+        model = DataType
+
+        fields = ["id", "name", "parent", "description", "projects"]
+
+    projects = serializers.SerializerMethodField()
+
+    def get_projects(self, obj):
+        """
+        Get the other sources this project requests access to.
```
Comment (madprime) [inline private_sharing/serializers.py]: incorrect due to copy/paste?

File: private_sharing/serializers.py [hunk 4]
```diff
@@ -116,3 +116,23 @@ def to_representation(self, obj):
...
+    """
+
+    class Meta:  # noqa: D101
+        model = DataType
+
+        fields = ["id", "name", "parent", "description", "projects"]
+
+    projects = serializers.SerializerMethodField()
+
+    def get_projects(self, obj):
```
Comment (madprime) [inline private_sharing/serializers.py]: This needs to distinguish between source_projects and requesting_projects. (It's just that so far only source has been implemented, but the intent is to include requesting ones as well.)

File: private_sharing/api_views.py [hunk 5]
```diff
@@ -428,3 +441,23 @@ def post(self, request):
...
+    Lists the datatypes available and which projects use them.
+    """
+
+    serializer_class = DataTypeSerializer
+
+    def get_queryset(self):
+        """
+        Get the queryset and filter on project if provided.
+        """
+        project_label = self.request.GET.get("project", None)
```
Comment (madprime) [inline private_sharing/api_views.py]: should distinguish between source project and requesting project (this is source)

File: private_sharing/api_views.py (hunk 1 above)
Comment (jhdulaney) [reply private_sharing/api_views.py]: Roger

File: private_sharing/serializers.py (hunk 4 above)
Comment (jhdulaney) [reply private_sharing/serializers.py]: Roger that

**Pull request merged**